cmake/libappimageConfig.cmake.in

```diff
@@ -11,18 +11,12 @@ get_filename_component(LIBAPPIMAGE_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 # Our library dependencies (contains definitions for IMPORTED targets)
 include("${LIBAPPIMAGE_CMAKE_DIR}/libappimageTargets.cmake")
 
-# TODO: while we check here for the libs with pkgconfig,
-# we do not create here the respective targets as used
-# in the link interface of the exported libappimage* targets
-# Those target names need to be either mapped here to those
-# pkg_check_modules(IMPORTED_TARGET) creates or changed
-# in the libappimage buildsystem to the original ones.
 find_package(PkgConfig REQUIRED)
-pkg_check_modules(GLIB REQUIRED glib-2.0)
-pkg_check_modules(GIO REQUIRED gio-2.0)
-pkg_check_modules(ZLIB REQUIRED zlib)
-pkg_check_modules(CAIRO REQUIRED cairo)
-pkg_check_modules(OPENSSL REQUIRED openssl)
+pkg_check_modules(glib glib-2.0>=2.40 IMPORTED_TARGET)
```

Comment:
I thought it was a really bad idea to use imported targets in this context? The targets pollute the global targets namespace. The variables that are created by that module won't escape the scope of that file, though.

Reply:
I thought the same, but you really need to export those targets in order to allow the libappimage target to properly find the libraries that are required. Otherwise it will fail to link.
Also the target names must be the same that the ones used on `target_link_libraries(libappimage ...`

You can use the designed test case.

Reply:
Just including the libraries names in the `INTERFACE_LINK_LIBRARIES` is not enough you actually need the targets to be resolved in the client projects. That's why I got from testing the different options. But fell free to fork the branch an make your own tests.

Reply:
I see. Quite interesting. Looks like you tried out all ideas that came to my mind, too. Alright, accepted.